iron-request.html

```diff
@@ -440,6 +440,23 @@
     _updateStatus: function() {
       this._setStatus(this.xhr.status);
       this._setStatusText((this.xhr.statusText === undefined) ? '' : this.xhr.statusText);
+    },
+
+    /**
+     * Returns the given response header in way that won’t cause an error during
+     * CORS requests if it is not returned or accessible due to
+     * `Access-Control-Expose-Headers`.
+     *
+     * @param {string} header The HTTP response header to return.
+     * @return {string|null} The header value or null if it is not present.
+     */
+    _responseHeader: function(header) {
+      var allHeaders = this.xhr.getAllResponseHeaders();
+      if (!!allHeaders &&
+          allHeaders.toLowerCase().indexOf(header.toLowerCase()) !== -1) {
```

Comment:
So, looking at the spec, ironically we don't get access to the `Access-Control-Expose-Headers` header by default, so we can't actually reason on whether we can access the header without hitting `getAllResponseHeaders`.

This is adding a bunch of complexity to the code, and additional processing to every response handler.

I'd propose that we drop support for `X-JSON-Prefix`. It's new enough that removing it won't break anyone, and it's just more complex than its worth. 

Sorry about asking you to remove it after I was the one who requested it in the first place!